numpyro/distributions/discrete.py

```diff
@@ -1,171 +1,393 @@
+
+    @property
+    def variance(self):
+        return np.broadcast_to(np.expand_dims(self.total_count, -1) * self.probs * (1 - self.probs),
+                               self.batch_shape + self.event_shape)
+
+    @property
+    def support(self):
+        return constraints.multinomial(self.total_count)
+
+
+class Categorical(Distribution):
+    arg_constraints = {'probs': constraints.simplex}
+
+    def __init__(self, probs, validate_args=None):
+        if np.ndim(probs) < 1:
+            raise ValueError("`probs` parameter must be at least one-dimensional.")
+        self.probs = probs
+        super(Categorical, self).__init__(batch_shape=np.shape(self.probs)[:-1],
+                                          validate_args=validate_args)
+
+    def sample(self, key, size=()):
+        return categorical_rvs(key, self.probs, shape=size + self.batch_shape)
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        batch_shape = lax.broadcast_shapes(np.shape(value), self.batch_shape)
+        value = np.expand_dims(value, axis=-1)
+        value = np.broadcast_to(value, batch_shape + (1,))
+        log_pmf = np.broadcast_to(self.logits, batch_shape + np.shape(self.logits)[-1:])
+        return np.take_along_axis(log_pmf, value, axis=-1)[..., 0]
+
+    @lazy_property
+    def logits(self):
+        return _to_logits_multinom(self.probs)
+
+    @property
+    def mean(self):
+        return np.broadcast_to(np.nan, self.batch_shape)
+
+    @property
+    def variance(self):
+        return np.broadcast_to(np.nan, self.batch_shape)
+
+    @property
+    def support(self):
+        return constraints.integer_interval(0, np.shape(self.probs)[-1])
+
+
+class CategoricalWithLogits(Distribution):
+    arg_constraints = {'logits': constraints.real}
+
+    def __init__(self, logits, total_count=1, validate_args=None):
+        if np.ndim(logits) < 1:
+            raise ValueError("`logits` parameter must be at least one-dimensional.")
+        logits = logits - logsumexp(logits)
+        self.logits = logits
+        super(CategoricalWithLogits, self).__init__(batch_shape=np.shape(logits)[:-1],
+                                                    validate_args=validate_args)
+
+    def sample(self, key, size=()):
+        return categorical_rvs(key, self.probs, shape=size + self.batch_shape)
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        value = np.expand_dims(value, -1)
+        value, log_pmf = promote_shapes(value, self.logits)
+        value = value[..., :1]
+        return np.take_along_axis(log_pmf, value, -1)[..., 0]
+
+    @lazy_property
+    def probs(self):
+        return _to_probs_multinom(self.logits)
+
+    @property
+    def mean(self):
+        return np.broadcast_to(np.nan, self.batch_shape)
+
+    @property
+    def variance(self):
+        return np.broadcast_to(np.nan, self.batch_shape)
```

Comment:
nit: no need for `np.broadcast_to`.

Reply:
We need the shape to be the same as `batch_shape` but we can use `lax.full` too. Will change.

Reply:
>"< I don't know why I make a comment for it here. Srr.